File: LeetCode/Medium/0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cs
public class Solution {
    public bool IncreasingTriplet(int[] nums) {
        var min = int.MaxValue;
        var mid = int.MaxValue;

        foreach (var n in nums)
        {
            if (n <= min)
            {
                min = n;
            }
            else if (n <= mid)
            {
                mid = n;
            }
            else
            {
                return true;
            }
        }

        return false;
    }
}

File: LeetCode/Medium/0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        if len(nums) < 3:
            return False
        num_1 = float("inf")
        num_2 = float("inf")

        for n in nums:
            if num_1 >= n:
                num_1 = n
            elif num_2 >= n:
                num_2 = n
            elif n > num_2:
                return True

        return False
        